_posts/2016-04-20-squabble-comments.md

```diff
@@ -0,0 +1,67 @@
+
+This is somewhat mitigated by this [semi-hidden setting](https://help.disqus.com/customer/portal/articles/1657951) that lets you opt out of some of their tracking, but I'm willing to venture that the vast majority of Disqus users don't know about this. And the "non-personally identifiable information" term is a [bit of a misnomer](https://www.eff.org/deeplinks/2009/09/what-information-personally-identifiable) in any case.
+
+(Full disclosure: we're not entirely innocent in this respect either, since we're using Google Analytics on our site, so if you are not doing so already, we invite you to go install an ad blocker plugin for your browser and come back to finish the article!)
+
+## Creating an alternative
+
+The privacy concerns and content ownership issues with Disqus were enough to make us look elsewhere for other solutions, and when the alternatives weren't exactly what we were looking for, we decided to roll our own.[^2]
+
+Our requirements were simple: allow users to leave a comment on a post, have that comment show up to other users who visit the post.
+
+The basic plan for implementing our Disqus alternative was similar to the Disqus model: on the Savas Labs website, embed some JavaScript in our page template; that JavaScript is then responsible for (1) loading any comments housed in our comment application, and (2) posting new comments to the comment application. The comment application, in turn, is a PHP application with an API that provides create/read/delete operations to users of our website.
+
+### Built on Lumen
+
+Looking around the open source landscape for the right tool to build this app, we came across [Lumen](https://lumen.laravel.com/), a slimmed down version of the well-known [Laravel](https://www.laravel.com) project. Lumen was very simple to get up and running with thanks to the [clear documentation examples](https://lumen.laravel.com/docs/5.2).
+
+The entry point of our application are some [routing rules](https://github.com/savaslabs/squabble/blob/master/app/Http/routes.php) that determine what data to give back to the user when they visit a particular API endpoint.
+
+From there, the requests are processed in the [CommentController](https://github.com/savaslabs/squabble/blob/master/app/Http/Controllers/CommentController.php) — depending on the request, we can create a comment, retrieve a comment, delete it, or get a quick list of all posts and the number of comments per page (we use this information on our home page and on the [News](/news) page to show the comment count per article).
+
+The application runs on one of our cloud servers on Linode, and uses a SQLite backend. Because the Lumen framework is very lean, and the application does relatively little, it's very fast — the `api/comments/count` call takes ~84 ms while loading all comments (`api/comments`) takes about 125 ms.
```

Comment:
I think it should be "an SQLite backend" rather than "a SQLite backend". I wasn't sure, but I checked the "a" vs. "an" rule and it's based on the first sound rather than the first letter so "es-q-lite". Reference: "For example, let’s consider the word “hour.” It starts with an “h,” which is a consonant. However, the “h” in “hour” is silent; therefore, the first sound from this word is a vowel sound." http://www.everythingenglishblog.com/?p=541